File: sdk/python/brikk/resources/health.py
# -*- coding: utf-8 -*-
"""Health check resources."""

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .._http import HTTPClient

from ..types import HealthStatus


class HealthResource:
    """Health check operations."""

    def __init__(self, http_client: "HTTPClient"):
        self._http = http_client

    def ping(self) -> HealthStatus:
        """Check if the API is responsive.

        Returns:
            Health status dict with 'status' field

        Example:
            >>> client.health.ping()
            {'status': 'ok'}
        """
        return self._http.get("/healthz")

    def readiness(self) -> HealthStatus:
        """Check if the API is ready to serve requests.

        Returns:
            Readiness status dict

        Example:
            >>> client.health.readiness()
            {'status': 'ready'}
        """
        return self._http.get("/readyz")

    def coordination_health(self) -> HealthStatus:
        """Check coordination service health.

        Returns:
            Coordination service health status

        Example:
            >>> client.health.coordination_health()
            {'status': 'ok', 'service': 'coordination'}
        """
        return self._http.get("/api/v1/coordination/health")

File: sdk/node/src/resources/health.ts
/**
 * Health check resources.
 */

import { HTTPClient } from '../http.js';

export interface HealthStatus {
  status: string;
  timestamp?: string;
  version?: string;
}

export class HealthResource {
  constructor(private http: HTTPClient) {}

  /**
   * Check if the API is responsive.
   */
  async ping(): Promise<HealthStatus> {
    return this.http.get<HealthStatus>('/healthz');
  }

  /**
   * Check if the API is ready to serve requests.
   */
  async readiness(): Promise<HealthStatus> {
    return this.http.get<HealthStatus>('/readyz');
  }

  /**
   * Check coordination service health.
   */
  async coordinationHealth(): Promise<HealthStatus> {
    return this.http.get<HealthStatus>('/api/v1/coordination/health');
  }
}

